api/app/controllers/spree/api/v2/base_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Spree
+  module Api
+    module V2
+      class BaseController < ActionController::API
+        include CanCan::ControllerAdditions
+
+        private
+
+        def spree_current_store
+          @spree_current_store ||= Spree::Store.current(request.env['SERVER_NAME'])
+        end
+
+        def spree_current_user
+          @spree_current_user ||= Spree.user_class.find_by(id: doorkeeper_token.resource_owner_id) if doorkeeper_token
+        end
+
+        def spree_authorize!(action, subject, *args)
+          authorize!(action, subject, *args)
+        end
+
+        # Needs to be overriden so that we use Spree's Ability rather than anyone else's.
+        def current_ability
+          @current_ability ||= Spree::Ability.new(spree_current_user)
+        end
+
+        def order_token
+          request.headers['X-Spree-Order-Token'] || params[:order_token]
+        end
+
+        def current_order
+          @current_order || find_or_create_current_order
+        end
+
+        def find_or_create_current_order
+          @current_order = Spree::CurrentOrderFinder.new(user: spree_current_user,
+                                                         store: spree_current_store,
+                                                         guest_token: order_token,
+                                                         currency: params[:currency] || current_currency).execute
+          if @current_order.nil?
+            @current_order = CreateOrder.new.call(user: spree_current_user, store: spree_current_store)
+          end
+
+          @current_order
+        end
+
+        def current_currency
+          Spree::Config[:currency]
```

Comment:
@bbonislawski I think we should get this value from the current store with a graceful fallback to `Spree::Config[:currency]` if there's no value set

so maybe ti should be:

```
spree_current_store.default_currency || Spree::Config[:currency]
```
?

Reply:
We decided to leave handling currency from current_store for some feature release. I think this can be good occassion 👍